actions-test-infra/deploy-cloud-function/sa.tf

```
@@ -53,6 +58,22 @@ resource "google_service_account_iam_member" "cf-default-sa" {
   member             = "serviceAccount:${google_service_account.deploy-cf-it-sa.email}"
 }
 
+# Assign the IAM Service Account User role on for Cloud Build to impersonate the
+# runtime service account.
+resource "google_service_account_iam_member" "cf-impersonate-sa" {
+  service_account_id = "projects/-/serviceAccounts/${google_service_account.deploy-cf-it-sa.email}"
+  role               = "roles/iam.serviceAccountUser"
+  member             = "serviceAccount:${data.google_project.project.number}@cloudbuild.gserviceaccount.com"
+}
+
+# Assign the IAM Service Account User role for the SA to impersonate itself
+# (yea, really).
+resource "google_service_account_iam_member" "self-impersonate-sa" {
+  service_account_id = "projects/-/serviceAccounts/${google_service_account.deploy-cf-it-sa.email}"
```

nit: I believe we can just do
```suggestion
  service_account_id = google_service_account.deploy-cf-it-sa.id
```